File: src/net/eduard/api/server/PartySystem.java
package net.eduard.api.server;

import java.util.List;

import net.eduard.api.lib.modules.FakePlayer;

public interface PartySystem {

	public boolean isInParty(FakePlayer player);

	public boolean isLeader(FakePlayer player);

	public boolean isMember(FakePlayer player);
	
	public FakePlayer getPartyLeader(FakePlayer player);

	public List<FakePlayer> getPartyMembers(FakePlayer player);

}

File: src/main/java/net/eduard/api/server/PartySystem.kt
package net.eduard.api.server

import net.eduard.api.lib.command.PlayerOffline
import net.eduard.api.server.party.Party
import net.eduard.api.server.party.PartyPlayer

interface PartySystem {

    fun isInParty(player: PlayerOffline): Boolean

    fun isLeader(player: PlayerOffline): Boolean

    fun isMember(player: PlayerOffline): Boolean

    fun getPartyLeader(player: PlayerOffline): PlayerOffline

    fun getPartyMembers(player: PlayerOffline): List<PlayerOffline>

    fun getParty(player : PlayerOffline) : Party

    fun getPlayer(player : PlayerOffline) : PartyPlayer
}
